### PR TITLE
Android 3.x Gradle, Automation , github and release script changes

### PR DESCRIPTION
**Pull request by PravinPK**
- Checkstyle
- Spotless
has been disabled to run for CI.
- WIll enable them in my next PR

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,93 +12,126 @@
...
+
+      # - run:
+      #     name: Check Style
+      #     command: make checkstyle
+
+      - android/save-gradle-cache
+
+      - android/save-build-cache
+
+      # Store Lint report
```
Comment (cacheung) [inline .circleci/config.yml:57]: we can remove the store link report comment.

File: code/target/consumer-rules.pro [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (cacheung) [inline code/target/consumer-rules.pro:1]: Do we still need this file?

File: scripts/version.sh [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# Make this script executable from terminal:
+# chmod 755 version.sh
+set -e # Any subsequent(*) commands which fail will cause the shell script to exit immediately
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+LINE="================================================================================"
+VERSION_REGEX="[0-9]+\.[0-9]+\.[0-9]+"
+
+GRADLE_PROPERTIES_FILE=$ROOT_DIR"/code/gradle.properties"
+CONSTANTS_FILE=$ROOT_DIR"/code/target/src/main/java/com/adobe/marketing/mobile/target/Target.java"
```
Comment (cacheung) [inline scripts/version.sh:12]: Should be "/code/target/src/phone/java/com/adobe/marketing/mobile/Target.java"?

File: .circleci/config.yml (hunk 1 above)
Comment (PravinPK) [reply .circleci/config.yml:57]: yeap 👍🏼  will do

File: scripts/version.sh (hunk 3 above)
Comment (PravinPK) [reply scripts/version.sh:12]: good call

**Review by cacheung — APPROVED** [review]

**Pull request merged**